safekeeper/src/metrics.rs

```diff
@@ -204,6 +204,28 @@ pub static WAL_BACKUP_TASKS: Lazy<IntCounterPair> = Lazy::new(|| {
     )
     .expect("Failed to register safekeeper_wal_backup_tasks_finished_total counter")
 });
+pub static WAL_RECEIVERS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
+        "safekeeper_wal_receivers",
+        "Number of currently connected WAL receivers (i.e. connected computes)"
+    )
+    .expect("Failed to register safekeeper_wal_receivers")
+});
+pub static WAL_RECEIVER_QUEUE_DEPTH: Lazy<Histogram> = Lazy::new(|| {
+    // Use powers of two buckets, but add a bucket at 0 and the max queue size to track empty and
+    // full queues respectively.
+    let mut buckets = pow2_buckets(1, MSG_QUEUE_SIZE);
+    buckets.insert(0, 0.0);
+    buckets.insert(buckets.len() - 1, (MSG_QUEUE_SIZE - 1) as f64);
```

Comment:
A bit unsure about the buckets here.

Powers of two gives poor resolution at the top -- we could consider e.g. linear buckets of 32, either from 0 or powers-of-two up to 64 then linear. But it's probably fine.

Inserting a bucket at 255 is also perhaps a bit odd. The motivation was to get a separate bucket for full queues. We could also consider shifting all buckets down by 1 (so 0,1,3,7,15,...255,+Inf) and/or using infinity for full queues.

Reply:
I think this is fine. We can tweak as we go.